File: Java/Software Development Module - Java/Revision/Exam2015/EncoderApp.java
/*
EncoderApp.java
Billy Cussen
10/08/2020
*/

import java.util.Scanner;
import java.util.InputMismatchException;

public class EncoderApp{

    public static void main(String [] args){

        //Variables
        String input;
        String output;
        boolean inputCheck;

        //Objects
        Scanner keyboard;
        Encoder myEncoder;

        do{
            input = new String();
            keyboard = new Scanner(System.in);
            inputCheck = true;

            try{
                System.out.print("Please enter your Input Sentance: ");
                input = keyboard.nextLine();

                if(input.length() <= 5){
                    throw new InputMismatchException("Your Sentance is too Short to Encode!");
                }

                for(int i = 0; i < input.length(); i++){

                    char ch = Character.toLowerCase(input.charAt(i));
                    if(!((ch >= 'a' || ch <= 'z') || ch == ' ' || ch == '.')|| Character.isDigit(ch)){
                        throw new InputMismatchException("Your Sentance can only contain Letters, Spaces and Full Stops!");
                    }

                }

            }catch(InputMismatchException e){
                inputCheck = false;
                System.out.println(e.getMessage());
            }finally{
                keyboard = null;
            }

        }while(!inputCheck);

        myEncoder = new Encoder();
        output = myEncoder.getEncodedSentance(input);
        myEncoder = null;
        System.out.println(output);

    }

}

File: Python/Software Development Module - Python/Revision/Exam2015/EncoderApp.py
"""
EncoderApp.py
Billy Cussen
28/10/2020
"""

#Object
import gc
import Encoder
from Encoder import myEncoder

#Variables
repeat = "y"

while repeat == "y":
    inputCheck = False

    #Input - Sentance
    while inputCheck == False:
        try: 
            sentance = str(input("Please enter the Sentance you would like to Encode: "))
            if (all(x.isalpha() or x.isspace() or x == "." for x in sentance) == False) or len(sentance) <= 5:
                raise Exception
        except Exception as e:
            print("You need to enter a Sentance greater than 5 characters with letters, spaces and full stops only!")
        else:
            inputCheck = True

    #Process & Output
    myEncoderApp = myEncoder(sentance)
    print(myEncoderApp.encodeSentance())
    myEncoderApp = None
    gc.collect()

    #Input - Repeat?
    repeat = ""
    while repeat != "y" and repeat != "n":
        repeat = str(input("Please enter \"y\" to retry, \"n\" to exit: "))
        repeat = repeat.lower()
        if repeat != "y" and repeat != "n":
            print("You need to enter \"y\" or \"n\"")

print("Thanks for using my Encoder App!")